test/infrastructure/docker/examples/simple-cluster-without-kcp.yaml

```diff
@@ -68,40 +68,51 @@ spec:
         eviction-hard: nodefs.available<0%,nodefs.inodesFree<0%,imagefs.available<0%
 ---
 apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
-kind: DockerMachine
+kind: DockerMachineTemplate
 metadata:
-  name: worker-0
+  name: worker
   namespace: default
+spec:
+  template:
+    spec: {}
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
+kind: KubeadmConfigTemplate
+metadata:
+  name: worker
+spec:
+  template:
+    spec:
+      joinConfiguration:
+        nodeRegistration:
+          kubeletExtraArgs:
+            eviction-hard: nodefs.available<0%,nodefs.inodesFree<0%,imagefs.available<0%
 ---
 apiVersion: cluster.x-k8s.io/v1alpha3
-kind: Machine
+kind: MachineDeployment
 metadata:
+  name: worker-md-0
   labels:
     cluster.x-k8s.io/cluster-name: my-cluster
-  name: worker-0
-  namespace: default
 spec:
-  version: "v1.14.2"
   clusterName: my-cluster
-  bootstrap:
-    configRef:
-      apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
-      kind: KubeadmConfig
-      name: worker-0-config
-      namespace: default
-  infrastructureRef:
-    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
-    kind: DockerMachine
-    name: worker-0
-    namespace: default
----
-apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
-kind: KubeadmConfig
-metadata:
-  name: worker-0-config
-  namespace: default
-spec:
-  joinConfiguration:
-    nodeRegistration:
-      kubeletExtraArgs:
-        eviction-hard: nodefs.available<0%,nodefs.inodesFree<0%,imagefs.available<0%
+  replicas: 1
+  selector:
+    matchLabels:
+      cluster.x-k8s.io/cluster-name: my-cluster
```

Comment:
nit: I think we can remove this label given it is automatically created by CAPI
(same in the template)
WDYT?

Reply:
@fabriziopandini yea, let me make the change.

Reply:
Looks like the label under `matchLabels` is required. I removed the redundant ones in `MachineDeployment.Metadata` and `MachineDeployment.Spec.Template`. Please help take a look. Thanks !
@fabriziopandini